### PR TITLE
feat: Add bulk import queries

### PR DESCRIPTION
**Pull request by anku255**
## Summary of change
(A few sentences about this PR)

## Related issues

- https://github.com/supertokens/supertokens-core/issues/912

## Test Plan
(Write your test plan here. If you changed any code, please provide us with clear instructions on how you verified your changes work. Bonus points for screenshots and videos!)

## Documentation changes
(If relevant, please create a PR in our [docs repo](https://github.com/supertokens/docs), or create a checklist here highlighting the necessary changes)

## Checklist for important updates
- [ ] Changelog has been updated
- [ ] `pluginInterfaceSupported.json` file has been updated (if needed)
- [ ] Changes to the version if needed
   - In `build.gradle`
- [ ] Had installed and ran the pre-commit hook
- [ ] If there are new dependencies that have been added in `build.gradle`, please make sure to add them in `implementationDependencies.json`.
- [ ] Issue this PR against the latest non released version branch.
   - To know which one it is, run find the latest released tag (`git tag`) in the format `vX.Y.Z`, and then find the latest branch (`git branch --all`) whose `X.Y` is greater than the latest released tag.
   - If no such branch exists, then create one from the latest released branch.
- [ ] When adding new recipes, ensure that its performance is being measured in the `OneMillionUsersTest`

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -2,7 +2,7 @@ plugins {
     id 'java-library'
 }
 
-version = "7.0.0"
+version = "7.1.0"
```
Comment (sattvikc) [inline build.gradle:6]: Update base branch with latest and update versions

File: src/main/java/io/supertokens/storage/mysql/queries/BulkImportQueries.java [hunk 2]
```diff
@@ -0,0 +1,303 @@
...
+        String tableName = Config.getConfig(start).getBulkImportUsersTable();
+        return "CREATE TABLE IF NOT EXISTS " + tableName + " ("
+                + "id CHAR(36),"
+                + "app_id VARCHAR(64) NOT NULL DEFAULT 'public',"
+                + "primary_user_id VARCHAR(64),"
+                + "raw_data TEXT NOT NULL,"
+                + "status VARCHAR(128) DEFAULT 'NEW',"
+                + "error_msg TEXT,"
+                + "created_at BIGINT NOT NULL, "
+                + "updated_at BIGINT NOT NULL,"
```
Comment (sattvikc) [inline src/main/java/io/supertokens/storage/mysql/queries/BulkImportQueries.java:52]: you are doing EPOCH default in postgres

File: src/main/java/io/supertokens/storage/mysql/queries/BulkImportQueries.java [hunk 3]
```diff
@@ -0,0 +1,303 @@
...
+        }
+
+        update(start, queryBuilder.toString(), pst -> {
+            int parameterIndex = 1;
+            for (BulkImportUser user : users) {
+                pst.setString(parameterIndex++, user.id);
+                pst.setString(parameterIndex++, appIdentifier.getAppId());
+                pst.setString(parameterIndex++, user.toRawDataForDbStorage());
+                pst.setLong(parameterIndex++, System.currentTimeMillis());
+                pst.setLong(parameterIndex++, System.currentTimeMillis());
```
Comment (sattvikc) [inline src/main/java/io/supertokens/storage/mysql/queries/BulkImportQueries.java:96]: please try to be consistent with this between postgres and mysql. I'm fine if you want to remove default in postgress and add these 2 lines there.

**Review by sattvikc — CHANGES_REQUESTED** [review]

File: src/main/java/io/supertokens/storage/mysql/queries/BulkImportQueries.java [hunk 4]
```diff
@@ -0,0 +1,303 @@
...
+        });
+    }
+
+    public static List<String> deleteBulkImportUsers(Start start, AppIdentifier appIdentifier,
+            @Nonnull String[] bulkImportUserIds) throws SQLException, StorageQueryException {
+        if (bulkImportUserIds.length == 0) {
+            return new ArrayList<>();
+        }
+
+        String selectQuery = "SELECT id FROM " + Config.getConfig(start).getBulkImportUsersTable()
```
Comment (rishabhpoddar) [inline src/main/java/io/supertokens/storage/mysql/queries/BulkImportQueries.java:223]: add a comment explaining why you are doing select first

**Review by rishabhpoddar — CHANGES_REQUESTED** [review]

File: src/main/java/io/supertokens/storage/mysql/queries/BulkImportQueries.java (hunk 3 above)
Comment (anku255) [reply src/main/java/io/supertokens/storage/mysql/queries/BulkImportQueries.java:96]: I have updated the postgres plugin to use the same code.

File: src/main/java/io/supertokens/storage/mysql/queries/BulkImportQueries.java (hunk 4 above)
Comment (anku255) [reply src/main/java/io/supertokens/storage/mysql/queries/BulkImportQueries.java:229]: done

**Pull request closed without merging**